common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorManager.kt

```diff
@@ -66,6 +67,15 @@ interface SensorManager {
         return sensor.enabled
     }
 
+    // Request to update a sensor, including any broadcast intent which may have triggered the request
+    // The intent will be null if the update is being done on a timer, rather than as a result
+    // of a broadcast being received.
+    fun requestSensorUpdate(context: Context, intent: Intent?) {
+        // Few sensors care about the intent, so allow them to just implement the interface that
+        // does not get passed that parameter.
+        requestSensorUpdate(context)
+    }
+
     fun requestSensorUpdate(context: Context)
```

Comment:
Rather than create a whole new function, why not just give an optional parameter?
```suggestion
    fun requestSensorUpdate(context: Context, intent: Intent? = null)
```

Reply:
This is the abstract definition of a funtion in an interface, so I don't think that adding an optional parameter acts as you are implying.  If I changed the abstract function, I would then have to add the new parameter to each of the existing concrete classes that implement the interface (about 25 files), and as a breaking change, this would also break every private fork of Home Assistant that has implemented its own Sensor classes.

By instead adding a new function to the interface, with a concrete body, no changes are needed to existing implementation classes, but new classes can chose to implement the new method if they need to.

Reply:
For now we can leave it, however I would argue we shouldn't worry about forks who have no interest in contributing upstream.  If we need a refactor, we will do it.